### DISCUSSION TITLE
ungroup empty folders

### DISCUSSION THREAD

**Discussion by juefeiyan** [nvim-tree/nvim-tree.lua#2645] [Q&A]
Hi there,
I want to request a new feature for the nvim-tree: when you open/expand grouped folders, you can ungroup them. The user case is I sometimes need to add files in the middle folder, for example, if the main class of my java project is in com/a/b/c/app/Main.java, I might need to add a folder under com/a/b/c/config, for now, I am not sure how to do that.

the logic would be:  grouped folders[closed] --> grouped folders[open] --> ungrouped folders[open] -->  grouped folders[closed] ...

If there is a way to do that, please tell me. If not, I am happy to implement this feature.

**Comment by gegoune** [comment]
You can simply remove unwanted paths from prompt, not ideal and this is still a valid feature request.

**Reply by juefeiyan** [reply]
Thank. Is it ok that i create a feature request and take that task?

**Reply by gegoune** [reply]
It's more than OK, PRs are very welcomed.
